src/Storages/MergeTree/MergeTreeDataPartCloner.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include "MergeTreeData.h"
+
+namespace DB
+{
+    /*
+     * Clones a source partition into a destination partition. This class was extracted from MergeTreeData::cloneAndLoadDataPartOnSameDisk
+     * as an attempt to make it a bit more readable and allow part of the code to be re-used. Reference method can be found in:
+     * https://github.com/ClickHouse/ClickHouse/blob/ee515b8862a9be0c940ffdef0a56e590b5facdb2/src/Storages/MergeTree/MergeTreeData.cpp#L5948.
```

Comment:
Interesting, maybe just `@see MergeTreeData::cloneAndLoadDataPartOnSameDisk` ?

Reply:
Well, that method will kind of cease to exist, right? Or maybe I am missing something

Reply:
URLs in the comments are not that useful, since you have to open them, and besides it does not give reader a context, i.e. method name. Plus there are no such references in the code right now.

It is true that this method can go away, but I would say that it is the responsibility of the author if that change. And even if this method will go away, it's name can still tell you something.

But maybe someone has other thoughts on this.

Reply:
That method's body will be replaced by the below, the only reason I didn't delete the method is to avoid breaking existing API and touching multiple un-related files. The change is already in this PR:

```
MergeTreeDataPartCloner cloner(params..);

return cloner.clone();
```
I don't see much value in referencing an "empty" method. In any case, I'll not argue over a comment/ docs. If you prfer this way, I am fine with that.